src/core/lv_refr.c

```diff
@@ -1113,6 +1113,10 @@ static void refr_obj(lv_layer_t * layer, lv_obj_t * obj)
             layer_area_act.y2 = layer_area_act.y1 + max_rgb_row_height - 1;
             if(layer_area_act.y2 > layer_area_full.y2) layer_area_act.y2 = layer_area_full.y2;
             bool area_need_alpha = alpha_test_area_on_obj(obj, &layer_area_act);
+
+            const void * bitmap_mask_src = lv_obj_get_style_bitmap_mask_src(obj, 0);
+            if(bitmap_mask_src) area_need_alpha  = true;
+
```

Comment:
Can we avoid calling `alpha_test_area_on_obj` calling it if we know `bitmap_mask_src != NULL`?

```suggestion
            const void * bitmap_mask_src = lv_obj_get_style_bitmap_mask_src(obj, 0);
            bool area_need_alpha = bitmap_mask_src || alpha_test_area_on_obj(obj, &layer_area_act);
```